docs/content/modules/multiwarehouse/inventory-module.md

```diff
@@ -0,0 +1,92 @@
+- `incoming_quantity`: a number indicating an incoming stock quantity of the item into the associated location. This attribute does not play into the `stocked_quantity` or when confirming whether a product variant is in stock.
+
+The `InventoryLevel` entity is associated with a location through the `location_id` attribute. There’s no direct relation between this `InventoryLevel` and a location entity, allowing for greater flexibility in what type of location that can be implemented. The core Medusa package assigns the value of `location_id` to the ID of an instance of the `StockLocation` entity, which is available within the [Stock Location module](./stock-location-module.md).
+
+### ReservationItem
+
+The `ReservationItem` entity represents a quantity of an inventory item that is reserved when an order is placed but not fulfilled yet. The process and importance of this is explained further in the [Order Placement section](#order-placement).
+
+The `ReservationItem` entity has the following notable attributes, among others:
+
+- `line_item_id`: The ID of the line item in the order that this reservation item refers to.
+- `inventory_item_id`: The ID of the inventory item this reservation item refers to.
+- `location_id`: The ID of the location that this item is reserved from. This is inferred from the sales channel when the order is placed, but can be edited by the admin.
+- `quantity`: A number indicating the quantity to be reserved.
+
+---
+
+## Inventory Process Overview
+
+### Product Variant Creation
+
+When a product variant is created, an inventory item is created along with the product variant automatically if the `manage_inventory` attribute in the variant is enabled. When the inventory item is created, it is attached to the product variant, which is represented by the `ProductVariantInventoryItem` entity as explained earlier.
+
+Inventory Levels are created when the admin sets the available quantity of a product variant in a stock location.
+
+### Cart and Checkout
+
+During the cart and checkout workflow, for example when a product inventory is added to the cart or during cart validation, the inventory module is used to confirm that an item has sufficient stock to be purchased in the desired quantity.
+
+As an inventory item can exist in multiple locations, the checking is done across those locations. The locations are retrieved based on the sales channel of the cart, as each location is associated with a sales channel.
+
+:::tip
+
+You can learn more about the relation between Stock Locations and Sales Channels in the [Stock Location documentation](./stock-location-module.md).
+
+:::
+
+Then, the inventory module confirms that the product variant has sufficient quantity across these locations by adding all the `stocked_quantity` of the inventory levels associated with these locations. When retrieving the `stocked_quantity` of each of the inventory levels, the `reserved_quantity` is subtracted from it to ensure accurate availability.
```

Comment:
should we add a note that if no inventory item exists for a variant it's assumed that it's available? just to explicitly highlight it for users who might surprised by this initially?

Reply:
Good point! I'll add it in